x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStoreTests.java

```diff
@@ -205,6 +210,80 @@ public void testPut() throws Exception {
         }
     }
 
+    /*
+     * This tests that we don't see deadlock if we throw a lot of data at the ILMHistoryStore quickly.
+     */
+    // @TestLogging(
+    // value = "org.elasticsearch.action.bulk:trace",
+    // reason = "Logging information about locks useful for tracking down deadlock"
+    // )
+    public void testDeadlock() throws Exception {
```

Comment:
Echoing our conversation on zoom -- can you rename this test? (/me ducks when you ask what a better name would be. 😄)